core/domain/topic_fetchers.py

```diff
@@ -0,0 +1,182 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.]
+
+"""Getter commands for for topic models."""
+
+import copy
+
+from core.domain import topic_domain
+from core.platform import models
+import feconf
+
+(topic_models,) = models.Registry.import_models([models.NAMES.topic])
+memcache_services = models.Registry.import_memcache_services()
+
+
+def _migrate_subtopics_to_latest_schema(versioned_subtopics):
+    """Holds the responsibility of performing a step-by-step, sequential update
+    of the subtopics structure based on the schema version of the input
+    subtopics dictionary. If the current subtopics schema changes, a
+    new conversion function must be added and some code appended to this
+    function to account for that new version.
+
+    Args:
+        versioned_subtopics: A dict with two keys:
+          - schema_version: int. The schema version for the subtopics dict.
+          - subtopics: list(dict). The list of dicts comprising the topic's
+              subtopics.
+
+    Raises:
+        Exception: The schema version of subtopics is outside of what
+            is supported at present.
+    """
+    subtopic_schema_version = versioned_subtopics['schema_version']
+    if not (1 <= subtopic_schema_version
+            <= feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION):
+        raise Exception(
+            'Sorry, we can only process v1-v%d subtopic schemas at '
+            'present.' % feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION)
+
+    while (subtopic_schema_version <
+           feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION):
+        topic_domain.Topic.update_subtopics_from_model(
+            versioned_subtopics, subtopic_schema_version)
+        subtopic_schema_version += 1
+
+
+def get_topic_memcache_key(topic_id, version=None):
+    """Returns a memcache key for the topic.
+
+    Args:
+        topic_id: str. ID of the topic.
+        version: int. The version of the topic.
+
+    Returns:
+        str. The memcache key of the topic.
+    """
+    if version:
+        return 'topic-version:%s:%s' % (topic_id, version)
+    else:
+        return 'topic:%s' % topic_id
+
+
+def get_new_topic_id():
```

Comment:
Should this function be in this file? I mean, this is only called before creating a topic and not related to when you want to fetch an existing topic from the datastore..

Reply:
I think we are using fetchers to get requests and this method as starts with `get`, as well as it acts as a new id fetcher.

Reply:
Should I move it back to services?

Reply:
Yes. This is more of a function along the "creation" flow lines. The _fetcher modules are meant to fetch the objects described.

Reply:
Done!